server/aesdsocket.c

```diff
@@ -189,192 +194,211 @@ void *timer_func(void *args)
 void *socketThreadProcessing(void *thread_params)
 {
 
-    thread_data *params = (thread_data *)thread_params;
-
-    char *client_read_buf = (char *)malloc(sizeof(char) * BUFFER_STD_SIZE);
-
-    if (client_read_buf == NULL)
+    thread_data* params = (thread_data*)thread_params;
+    
+    char* client_read_buf = (char*)malloc(sizeof(char) * BUFFER_STD_SIZE);
+    
+    if(client_read_buf == NULL)
     {
-        syslog(LOG_ERR, "malloc failed %d\n\r", (int)(params->thread));
-        params->thread_complete_status = true;
+       syslog(LOG_ERR,"malloc failed %d\n\r", (int)(params->thread));
+       params->thread_complete_status = true;
     }
     else
     {
-        memset(client_read_buf, 0, BUFFER_STD_SIZE);
+       memset(client_read_buf, 0, BUFFER_STD_SIZE);
     }
-
-    uint32_t counter = 1;
+    
+    uint32_t counter = 1; 
     int curr_pos = 0;
 
-    while (!(params->thread_complete_status))
+
+    while(!(params->thread_complete_status))
     {
-        int read_bytes = read(params->client_fd, client_read_buf + curr_pos, (BUFFER_STD_SIZE));
-        if (read_bytes < 0)
-        {
-            syslog(LOG_ERR, "reading from socket errno=%d\n", errno);
-            free(client_read_buf);
+	int read_bytes = read(params->client_fd, client_read_buf + curr_pos , (BUFFER_STD_SIZE));
+	if (read_bytes < 0) 
+	{
+	    syslog(LOG_ERR, "reading from socket errno=%d\n", errno);
+	    free(client_read_buf);
             params->thread_complete_status = true;
             pthread_exit(NULL);
-        }
-
-        if (read_bytes == 0)
+	}
+
+	if (read_bytes == 0)
+	{
+	    continue;
+	}
+	
+	curr_pos += read_bytes;
+
+	if (strchr(client_read_buf, '\n')) 
+	{  
+	    break; 
+	} 
+    
+	
+        counter++;
+        client_read_buf = (char*)realloc(client_read_buf, (counter * BUFFER_STD_SIZE));
+       
+        if(client_read_buf == NULL)
         {
-            continue;
+          syslog(LOG_ERR,"realloc error %d\n\r", (int)params->thread);
+          free(client_read_buf);
+          params->thread_complete_status = true;
+          pthread_exit(NULL);
         }
+    }
+    
+    
 
-        curr_pos += read_bytes;
+    int fd = open(FILE, O_RDWR | O_APPEND, 0644);
+    if (fd < 0)
+    {
+	syslog(LOG_ERR, "failed to open a file:%d\n", errno);
+    }
+    //Handling IOCTL, Preferrably write a separate func.
+    if(strncmp(client_read_buf, ioctl_aesdchar, strlen(ioctl_aesdchar))==0)
+    {
+        struct aesd_seekto seekto;
 
-        if (strchr(client_read_buf, '\n'))
+        sscanf(client_read_buf, "AESDCHAR_IOCSEEKTO:%d,%d",&seekto.write_cmd,&seekto.write_cmd_offset);
```

Comment:
@dhirajbennadi you need to check the return code here and do error handling, refer to discussion in class about this.